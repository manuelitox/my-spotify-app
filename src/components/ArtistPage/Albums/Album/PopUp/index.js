import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import Track from 'components/generic/Track/'
import PopUp from 'components/generic/PopUp/container'
import PlaceholderAlbum from 'svgs/placeholderAlbum.svg'
import StylesHeaderAlbum from './styles-header'
import StylesWrapperTracks from './styles-wrapper-tracks'

const PopUpAlbum = ({ isOpen, togglePopUp, cover, title, releaseDate, numberTracks }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp }>
    <Fragment>
      <StylesHeaderAlbum>
        <img 
          src={ cover ? cover : PlaceholderAlbum } 
          alt={ title }
          width="300"
          height="300" />
        <h3>{ title }</h3>
        <p>Total tracks: { numberTracks }</p>
        <p>Release date: { releaseDate }</p>
      </StylesHeaderAlbum>
      <StylesWrapperTracks>
        <Track 
          name="Empire State of Mind" 
          duration="3 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="No one" 
          duration="4 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />                
      </StylesWrapperTracks>
    </Fragment>
  </PopUp>      
)

PopUpAlbum.propTypes = {
  isOpen: PropTypes.bool,
  togglePopUp: PropTypes.func.isRequired,
  cover: PropTypes.string,
  title: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  numberTracks: PropTypes.number.isRequired
}

export default PopUpAlbum