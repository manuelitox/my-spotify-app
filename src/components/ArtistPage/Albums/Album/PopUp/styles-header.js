import styled from 'styled-components'

const StylesHeaderAlbum = styled.div`
  margin: 30px 0;
  img {
    @media only screen and (max-width: ${ props => props.theme.size.small }) {
      height: auto;
      max-width: 300px;
      width: 100%;
    }
  }
  h3 {
    color: ${ props => props.theme.blueDark };
    font-size: 1.6rem;
    font-weight: 500;
    margin: 10px 0 4px;
  }
  p {
    color: ${ props => props.theme.offWhite };
    font-size: 1.4rem;
    line-height: 1.6;
    margin: 0;
  }
`

export default StylesHeaderAlbum