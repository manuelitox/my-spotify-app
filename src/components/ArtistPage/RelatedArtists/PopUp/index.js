import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import theme from 'config/theme'
import PopUp from 'components/generic/PopUp/container'
import Title from 'components/ArtistPage/Title/'
import Artist from 'components/ArtistPage/RelatedArtists/Artist/'
import StyledWrapperRelatedArtists from 'components/ArtistPage/RelatedArtists/styles-wrapper'
import getImage from 'lib/get-image'

const PopUpRelatedArtists = ({ isOpen, togglePopUp, relatedArtists }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp }>
    <Fragment>
      <Title color={ theme.blueDark } insidePopUp={ true }>Related Artists</Title>
      <StyledWrapperRelatedArtists inPopup onClick={ togglePopUp }>
        { relatedArtists.map((artist, index) =>Â (
          <Artist key={ index } id={ artist.id } photo={ getImage(artist.images, 200) } name={ artist.name } />
        )) }   
      </StyledWrapperRelatedArtists>
    </Fragment>
  </PopUp>      
)

PopUpRelatedArtists.propTypes = {
  isOpen: PropTypes.bool,
  togglePopUp: PropTypes.func.isRequired,
  relatedArtists: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      images: PropTypes.array      
    })
  ).isRequired
}

export default PopUpRelatedArtists