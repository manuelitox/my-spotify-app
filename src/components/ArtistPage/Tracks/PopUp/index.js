import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import theme from 'config/theme'
import Title from 'components/ArtistPage/Title/'
import Track from 'components/generic/Track/'
import PopUp from 'components/generic/PopUp/container'
import StylesWrapperTracks from 'components/ArtistPage/stylesWrapperTracks'

const PopUpTopTracks = ({ isOpen, togglePopUp }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp }>
    <Fragment>
      <Title color={ theme.blueDark } insidePopUp={ true }>Top Tracks</Title>
      <StylesWrapperTracks>
        <Track 
          name="Empire State of Mind" 
          duration="3 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="No one" 
          duration="4 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />                
      </StylesWrapperTracks>
    </Fragment>
  </PopUp>      
)

PopUpTopTracks.propTypes = {
  isOpen: PropTypes.bool,
  togglePopUp: PropTypes.func.isRequired
}

export default PopUpTopTracks