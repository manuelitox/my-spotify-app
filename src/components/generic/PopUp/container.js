import React, { Component } from 'react'
import PropTypes from 'prop-types'
import onClickOutside from 'react-onclickoutside'

import PopUp from './'

export class PopUpContainer extends Component {
  handleClickOutside = () => this.props.togglePopUp()

  render () {
    return this.props.isOpen && <PopUp { ...this.props } />
  }
}

PopUpContainer.defaultProps = {
  WithCloseButton: true
}

PopUpContainer.propTypes = {
  width: PropTypes.string, 
  isOpen: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.string,
    PropTypes.object
  ]),
  togglePopUp: PropTypes.func.isRequired,
  WithCloseButton: PropTypes.bool
}

export default onClickOutside(PopUpContainer)