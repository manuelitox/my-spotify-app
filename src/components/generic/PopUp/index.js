import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import BodyClassName from 'react-body-classname'

import StyledPopUp from './styles'

const PopUp = ({ width, isOpen, children }) => (
  isOpen && <Fragment>
    <BodyClassName className="fixed" />
    <StyledPopUp width={ width }>
      { children }
    </StyledPopUp>
  </Fragment>
)

PopUp.propTypes = {
  width: PropTypes.string, 
  isOpen: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ])
}

export default PopUp