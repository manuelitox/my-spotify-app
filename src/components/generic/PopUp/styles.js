import styled from 'styled-components'
import PropTypes from 'prop-types'

const StyledPopUp = styled.div`
  background-color: white;
  height: 100vh;
  top: 0;
  overflow: auto;
  padding: 20px;
  position: absolute;
  right: 0;
  width: ${ props => props.width };
  z-index: 5;
  @media only screen and (max-width: ${ props => props.theme.size.small }) {
    width: 100%;
  }
`

StyledPopUp.defaultProps =Â {
  width: '500px'
}

StyledPopUp.propTypes = {
  width: PropTypes.string
}

export default StyledPopUp