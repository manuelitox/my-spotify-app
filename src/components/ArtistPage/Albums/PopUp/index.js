import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import theme from 'config/theme'
import PopUp from 'components/generic/PopUp/container'
import Title from 'components/ArtistPage/Title/'
import Album from 'components/ArtistPage/Albums/Album/'
import StyledWrapperAlbums from 'components/ArtistPage/Albums/styles'
import getImage from 'lib/get-image'

const PopUpAlbums = ({ isOpen, togglePopUp, albums, className }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp } className={ className }>
    <Fragment>
      <Title color={ theme.blueDark } insidePopUp={ true }>Albums</Title>
      <StyledWrapperAlbums inPopup>
        { albums.map( (album, index) => (
          <Album key={ index } id={ album.id } title={ album.name } cover={ getImage(album.images) } numberTracks={ album.total_tracks } releaseDate={ album.release_date } />
        )) }
      </StyledWrapperAlbums>
    </Fragment>
  </PopUp>      
)

PopUpAlbums.propTypes = {
  isOpen: PropTypes.bool,
  className: PropTypes.string,
  togglePopUp: PropTypes.func.isRequired,
  albums: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      total_tracks: PropTypes.number,
      release_date: PropTypes.string
    })
  )
}

export default PopUpAlbums