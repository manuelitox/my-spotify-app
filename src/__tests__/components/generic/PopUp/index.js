import React from 'react'
import { shallow } from 'enzyme'
import BodyClassName from 'react-body-classname'

import PopUp from 'components/generic/PopUp/'

describe('Component | PopUp', () => {
  let component 
  beforeEach( () => {
    component = shallow(
      <PopUp><p>PopUp!!</p></PopUp>
    )
  })
  it('should not render because isOpen is false by default', () => {
    expect(component.contains(<BodyClassName className="fixed" />)).toBeFalsy()
    expect(component.find('p').length).toEqual(0)
    expect(component).toMatchSnapshot()
  })
  it('should render a PopUp', () => {
    component = shallow(<PopUp isOpen><p>PopUp!!</p></PopUp>)
    expect(component.contains(<BodyClassName className="fixed" />)).toBeTruthy()
    expect(component.find('p').text()).toEqual('PopUp!!')
    expect(component).toMatchSnapshot()
  })
})