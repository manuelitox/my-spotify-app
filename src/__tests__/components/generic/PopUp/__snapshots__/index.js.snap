// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | PopUp should not render because isOpen is false by default 1`] = `""`;

exports[`Component | PopUp should render a PopUp 1`] = `
<Fragment>
  <SideEffect(BodyClassName)
    className="fixed"
  />
  <styled.div
    width="500px"
  >
    <p>
      PopUp!!
    </p>
  </styled.div>
</Fragment>
`;
